web/packages/teleterm/src/services/pty/ptyHost/buildPtyOptions.ts

```diff
@@ -46,9 +52,28 @@ export async function buildPtyOptions(
   cmd: PtyCommand
 ): Promise<{
   processOptions: PtyProcessOptions;
+  openedShell: Shell;
   creationStatus: PtyProcessCreationStatus;
 }> {
-  return resolveShellEnvCached(settings.defaultShell)
+  // Get the full shell object by the shell ID.
+  // We wouldn't have to do it if we sent a shell bin path from the renderer,
+  // but it's better to validate inputs that come from that process.
```

Comment:
Hmmm, but `buildPtyOptions` is itself called in the renderer process, no? I don't think we have to explain ourselves here, unless you want to highlight that we specifically pass only the ID of the shell so that it's easier to implement proper validation on the side of the shared process later on + a link to the issue.

Reply:
Yes, `buildPtyOptions` is called from renderer, but it gets `runtimeSettings` from preload, so it's a separate context. But it's true for other stuff in `buildPtyOptions` as well, so I will remove the comment.